File: ios/Utilities/CommonParams.swift
import Foundation

class CommonParams {
  private let params: NSDictionary?

  init(params: NSDictionary?) {
    self.params = params
  }

  private func getParam(_ key: String) -> Any? {
    guard let value = self.params?.value(forKey: key) else {
      return nil
    }

    return value
  }

  private func getParam(parent: String, key: String) -> String? {
    guard let parentValue = self.params?.value(forKey: parent) as? [String: String] else {
      return nil
    }

    return parentValue[key]
  }

  private func isDevelopment() -> Bool {
    guard let isDev: Bool = self.getParam("isDevelopment") as? Bool else {
      return false
    }

    return isDev
  }

  func isNull() -> Bool {
    if (self.params == nil) {
      return true
    }

    return self.params?.count == 0
  }

  func setHeaders(_ headers: NSDictionary) {
    Config.setHeaders(headers)
  }

  func build() {
    if self.isNull() {
      return
    }

    Config.setDeviceKeyIdentifier(self.getParam("deviceKeyIdentifier") as? String ?? "")
    Config.setBaseUrl(self.getParam("baseUrl") as? String ?? "")
    Config.setIsDevelopment(self.isDevelopment())
  }
}

File: examples/flutterapp/android/app/src/main/java/com/azifacemobile/utils/CommonParams.java
package com.azifacemobile.utils;

import androidx.annotation.Nullable;

import com.azifacemobile.Config;

import java.util.Map;

public class CommonParams {
    private final Map<String, Object> params;

    public CommonParams(Map<String, Object> params) {
        this.params = params;
    }

    private String getParam(String key) {
        if (this.params == null) return null;
        Object value = this.params.get(key);
        return value != null ? value.toString() : null;
    }

    private Boolean isDevelopment() {
        String isDev = this.getParam("isDevelopment");
        return isDev != null && isDev.equalsIgnoreCase("true");
    }

    public Boolean isNull() {
        return this.params == null;
    }

    public void setHeaders(@Nullable Map<String, Object> headers) {
        Config.setHeaders(headers);
    }

    public void build() {
        if (!this.isNull()) {
            Config.setDeviceKeyIdentifier(this.getParam("deviceKeyIdentifier"));
            Config.setBaseUrl(this.getParam("baseUrl"));
            Config.setIsDevelopment(this.isDevelopment());
        }
    }
}
